journal/src/main/java/io/camunda/zeebe/journal/file/SegmentedJournal.java

```diff
@@ -493,57 +496,110 @@ private FileChannel openChannel(final File file) {
   protected Collection<JournalSegment> loadSegments() {
     // Ensure log directories are created.
     directory.mkdirs();
+    final List<JournalSegment> segments = new ArrayList<>();
 
-    final TreeMap<Long, JournalSegment> segments = new TreeMap<>();
+    final List<File> files = getSortedLogSegments();
+    for (int i = 0; i < files.size(); i++) {
+      final File file = files.get(i);
 
-    // Iterate through all files in the log directory.
-    for (final File file : directory.listFiles(File::isFile)) {
+      try {
+        final JournalSegmentDescriptor descriptor = readDescriptor(file);
+        final JournalSegment segment = loadSegment(descriptor.id());
 
-      // If the file looks like a segment file, attempt to load the segment.
-      if (JournalSegmentFile.isSegmentFile(name, file)) {
-        final JournalSegmentFile segmentFile = new JournalSegmentFile(file);
-        final ByteBuffer buffer = ByteBuffer.allocate(JournalSegmentDescriptor.getEncodingLength());
-        try (final FileChannel channel = openChannel(file)) {
-          channel.read(buffer);
-          buffer.flip();
-        } catch (final IOException e) {
-          throw new JournalException(e);
+        if (i > 0 && segments.get(i - 1).lastIndex() + 1 != segment.index()) {
+          throw new CorruptedLogException(
+              String.format(
+                  "Log segment %s is not aligned with previous segment %s.",
```

Comment:
Add more context to the error message - last segment's last index, current segment index.